plugins/chunter-impl/src/components/CreateDirectMessage.svelte

```diff
@@ -44,32 +44,35 @@
   let accountUpdater: QueryUpdater<Account> | undefined = undefined
 
   $: accountUpdater = client.query<Account>(accountUpdater, core.class.Account, {}, (results) => {
-    allAccounts = results
+    allAccounts = results.filter(it => it._id !== client.accountId())
   })
 
+  let query: QueryUpdater<Message> | undefined = undefined
+
   $: client.findAll<Account>(core.class.Account, { _id: { $in: to } }).then((acc) => {
     toAccount = acc
-    currentSpace = undefined
     notifications = undefined
     messages = []
+    query?.unsubscribe()
+  })
 
-    // Check if we have already a channel between us.
-    client.findAll<Space>(chunter.class.Channel, { direct: true, private: true }).then((channels) => {
-      for (const c of channels) {
-        const m1 = [...c.members].sort()
-        const m2 = [client.accountId(), ...to].sort()
-        if (deepEqual(m1, m2)) {
-          // Ok, we have channel already
-          currentSpace = c
-          break
+  // Check if we have already a channel between us.
+  $: client.findAll<Space>(chunter.class.Channel, { direct: true, private: true }).then((channels) => {
+    for (const c of channels) {
+      const m1 = [...c.members].sort()
+      const m2 = [client.accountId(), ...to].sort()
```

Comment:
does it makes sense to sort it 1 time before the for loop?

Reply:
Thanks.